src/App.tsx

```diff
@@ -9,33 +9,70 @@ import { DetailAndCreateOption } from './Pages/DetailAndCreateOption';
 import { CreateQuestion } from './Pages/CreateQuestion';
 import { MyPage } from './Pages/MyPage';
 import { SolvingQuestion } from './Pages/SolvingQuestion';
+import styled from '@emotion/styled';
 
 function App() {
-  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(true)
+  return (
+    <Router>
+      <RoutingFromLoginState />
+    </Router>
+  )
+}
+
+function RoutingFromLoginState() {
+  const navigate = useNavigate();
+  //TODO) USING GLOBAL STATE BY REDUX
+  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
 
   useEffect(() => {
-    console.log(isLoggedIn)
-    setIsLoggedIn(true);
-  }, [])
+      if (!isLoggedIn) navigate("/login");
+      console.log(isLoggedIn)
+  }, [isLoggedIn])
+
+  function loginout(state: boolean){
+    setIsLoggedIn(state);
+  }
+
 
   return (
-    <div className='App'>
-      <Router>
-        <Gnb loginState={isLoggedIn}/>
-        <div className='Box'>
-        <Routes>
-          <Route path="/" element={<MainPage />} />
-          <Route path="/solve" element={<SolvingQuestion />} />
-          <Route path="/login" element={<LogIn />} />
-          <Route path="/enroll" element={<Enroll />}/>
-          <Route path="/createQuestion" element={<CreateQuestion />}/>
-          <Route path="/question/createOption" element={<DetailAndCreateOption />}/>
-          <Route path="/mypage" element={<MyPage stemNum={3} optionNum={4}/>}/>
-        </Routes>
-        </div>
-      </Router>
-    </div>
+    <Container>
+      <Gnb loginState={isLoggedIn}/>
+      <InnerBox>
+      <Routes>
+        <Route path="/" element={<MainPage />} />
+        <Route path="/solve" element={<SolvingQuestion />} />
+        <Route path="/login" element={<LogIn login={loginout}/>} />
+        <Route path="/enroll" element={<Enroll />}/>
+        <Route path="/createQuestion" element={<CreateQuestion />}/>
+        <Route path="/question/createOption" element={<DetailAndCreateOption />}/>
+        <Route path="/mypage" element={<MyPage logout={loginout} stemNum={3} optionNum={4}/>}/>
```

Comment:
I think `stemNum` and `optionNum` should be passed as url query params rather than component props!

Reply:
Yep it was just for showing in demo. 👌🏼